src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Tree/Dependencies/Subscriptions/DependencyRulesSubscriber.cs

```diff
@@ -109,19 +109,6 @@ IDisposable SyncLink((BufferBlock<IProjectVersionedValue<IProjectSubscriptionUpd
             IProjectSubscriptionUpdate projectUpdate = e.Item1;
             IProjectCatalogSnapshot catalogSnapshot = e.Item2;
 
-            // Broken design time builds sometimes cause updates with no project changes and sometimes
-            // cause updates with a project change that has no difference.
-            // We handle the former case here, and the latter case is handled in the CommandLineItemHandler.
```

Comment:
This comment regarding `CommandLineItemHandler` is a hint that this condition was copied over from an earlier version of the language service (if I remember the origins of the dependencies node code correctly) which likely had different constraints.

Reply:
Have you actually verified this works? I'm pretty sure that we have guards later than this that will prevent the new changes from being picked up. See the ones inside: https://github.com/dotnet/project-system/pull/6605.

Reply:
There's also assumptions later that evaluation rules always exist which will fail after this empty check,

Reply:
The checks you're referring to apply only to one kind of change. The production of a new snapshot (which prevents the leak) is also conditional on changes to the catalog. So long as we allow the change to continue here, a new catalog will be produced which addresses the leak.

> There's also assumptions later that evaluation rules always exist which will fail after this empty check

I'll investigate. It may be that only the second `if` here should be removed. In my testing I didn't see this case trigger, but then I didn't have a broken dtb.

Reply:
The comment is wrong, this will not trigger on broken design-time build (there will always be evaluation results in that case), it will trigger when the rules are missing due to bad import, for example.